pkgs/by-name/bu/bun/package.nix

```diff
@@ -44,7 +57,7 @@ stdenvNoCC.mkDerivation rec {
   postPatchelf =
     lib.optionalString stdenvNoCC.hostPlatform.isDarwin ''
       wrapProgram $out/bin/bun \
-        --prefix DYLD_LIBRARY_PATH : ${lib.makeLibraryPath [ darwin.ICU ]}
+        --prefix DYLD_LIBRARY_PATH : ${lib.makeLibraryPath [ icucore ]}
     ''
```

Comment:
I’m okay with doing this, but did `install_name_tool` not end up working?

I’d probably prefer leaving the binary alone over `DYLD_LIBRARY_PATH` shenanigans, since this still makes it possible for users of Bun to invoke external tools that are linked against Apple’s ICU and for havoc to result (because they target a newer OS version than our `darwin.ICU`, say). I don’t think supporting macOS 12 when upstream doesn’t is worth risking that kind of tricky‐to‐debug situation. But if `install_name_tool` works then I’m happy to go with that, as it’s probably not common to use Nixpkgs packages in a Hardened Runtime app anyway, so I think the entitlements shouldn’t matter.

Reply:
I never tried `install_name_tool`. I can switch it to `install_name_tool`, but that will also require re-signing the binary with an ad hoc signature (using either sigtool or rcodesign). This seemed like the easier change.

My primary concern is not causing a regression for users who needed https://github.com/NixOS/nixpkgs/pull/358195. We can revisit dropping old version workarounds once the minimum version is bumped for 25.11.